### ISSUE TITLE
Rule request: Prefer `.zero` over explicit init with zero args

### ISSUE THREAD

**Issue by MortyMerr** [realm/SwiftLint#3190]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

Please describe the rule idea, format
this issue's title as `Rule Request: [Rule Name]` and describe:

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

CoreGraphics types (`CGPoint`, `CGRect`, `CGSize`, `CGVector`) which are widely used in iOS development provide a convenient way to init object with `0` values (`.zero` property). It should be preferred over explicit init.

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.
```swift
// ok
CGPoint(x: 0, y: -1)
CGSize(width: 2, height: 4)
CGRect.zero

// triggers
CGPoint(x: 0, y: 0)
CGPoint(x: 0.000000, y: 0)
```
3. Should the rule be configurable, if so what parameters should be configurable?

Just severity.

4. Should the rule be opt-in or enabled by default? Why?

Opt-in.

**Comment by marcelofabri** [comment]
Implemented in #3191
